### PR TITLE
Feat/improve toolbar

### PR DESCRIPTION
**Pull request by bartvandebiezen**

### CODE REVIEW

**Review by tdjager — COMMENTED** [review]
## Recursive Review: PR #44 — Feat/improve toolbar

**Review agents**: 3 specialized (2× Opus, 1× Sonnet) + 1 Challenger (Opus) + 1 Validator (Opus)

**Verdict: REQUEST CHANGES** — One critical runtime crash must be fixed before merge. See inline comments for details.

File: src/components/lists-and-menus/menu/rr-menu.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	static override styles = menuDividerStyles;
+
+	override render() {
+		return menuDividerTemplate();
+	}
+}
+
+// # rr-menu-item
+
+@customElement('rr-menu-item')
```
Comment (tdjager) [inline src/components/lists-and-menus/menu/rr-menu.ts]: **CRITICAL: Duplicate `rr-menu-item` custom element registration — runtime crash**

This line registers `@customElement('rr-menu-item')`, but the same tag name is already registered in `src/components/navigation/menu-bar/rr-menu-item.ts:22`. Custom element names are global singletons — the second registration throws a `DOMException`.

Since `index.ts` imports both (line 75 exports the navigation version, line 80 exports `RRMenu` which triggers this registration as a side effect), any consumer using the barrel export will crash at runtime.

**Fix:** Rename this component to a distinct tag name, e.g. `rr-dropdown-menu-item` or `rr-action-menu-item`, and rename the class to avoid the `RRMenuItem` name collision as well.

File: src/components/index.ts [hunk 2]
```diff
@@ -70,26 +75,25 @@ export { RRMenuBar } from './navigation/menu-bar/rr-menu-bar.ts';
...
-export { RRDropDownFieldCell } from './lists/drop-down-field-cell/rr-drop-down-field-cell.ts';
-export { RRTextCell } from './lists/text-cell/rr-text-cell.ts';
-export { RRIconCell } from './lists/icon-cell/rr-icon-cell.ts';
-export { RRDescriptionCell } from './lists/description-cell/rr-description-cell.ts';
-export { RRStepperCell } from './lists/stepper-cell/rr-stepper-cell.ts';
-export { RRListItemDragHandle } from './lists/list-item-drag-handle/rr-list-item-drag-handle.ts';
-export { RRListItemDragHandleCell } from './lists/list-item-drag-handle-cell/rr-list-item-drag-handle-cell.ts';
-export { RRTimelineTrackCell } from './lists/timeline-track-cell/rr-timeline-track-cell.ts';
+// List and Menu components
+export { RRMenu } from './lists-and-menus/menu/rr-menu.ts';
```
Comment (tdjager) [inline src/components/index.ts:79]: **IMPORTANT: Missing named exports for new menu classes**

`RRMenu` is exported here, but importing it triggers the `rr-menu-item` side-effect registration that conflicts with line 75's `RRMenuItem` (from navigation/menu-bar). Additionally:
- The new `RRMenuItem` class (from lists-and-menus/menu) is never individually exported
- `RRMenuDivider` is never exported

After resolving the duplicate tag name (see comment on `rr-menu.ts:24`), add the renamed menu-item and divider classes to the barrel export.

File: src/components/lists-and-menus/menu/rr-menu.template.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	return html`
+		<div class="menu" role="menu" tabindex="-1">
+			<slot></slot>
+		</div>
+	`;
+}
+
+export function menuItemTemplate(this: RRMenuItem) {
+	return html`
+		<button
```
Comment (tdjager) [inline src/components/lists-and-menus/menu/rr-menu.template.ts]: **IMPORTANT: Missing ARIA roles for WCAG 2.1 AA compliance**

The menu container at line 6 has `role="menu"`, but these `<button>` elements inside `rr-menu-item` lack `role="menuitem"`. Per WAI-ARIA, children of `role="menu"` must have `role="menuitem"`, `role="menuitemcheckbox"`, or `role="menuitemradio"`.

Additionally:
- Selectable items should have `aria-checked=${this.selected}`
- The divider template should use `role="separator"`

**Fix:** Add `role="menuitem"` to this button element. For selectable items, also add `aria-checked`.

File: src/components/control-groups/toolbar/rr-toolbar.template.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+				${leftSpacerZero ? nothing : html`<div class="toolbar__left-spacer"></div>`}
+				${renderChildren(centerChildren, allChildren, overflowIds)}
+				${rightSpacerZero ? nothing : html`<div class="toolbar__right-spacer"></div>`}
+			` : isSoloFluidItem ? nothing : html`
+				<div class="toolbar__flexible-spacer"></div>
+			`}
+			${renderChildren(endChildren, allChildren, overflowIds)}
+		</div>
+		<div class="toolbar__overflow-button ${hasOverflow ? '' : 'is-hidden'}">
+			<rr-icon-button size=${size}>
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.template.ts]: **IMPORTANT: Missing `aria-haspopup` and `aria-label` on overflow button**

This overflow button opens a menu but lacks:
- `aria-haspopup="menu"` to indicate it opens a menu
- `aria-expanded` to communicate open/closed state

Also, the `role="toolbar"` on line 97 should have an `aria-label` so screen readers can distinguish multiple toolbars on a page. Consider adding an `aria-label` property to `rr-toolbar` and forwarding it.

File: src/components/lists-and-menus/menu/rr-menu.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+	@property({ type: Boolean, reflect: true })
+	disabled = false;
+
+	_handleClick(): void {
+		if (this.disabled) return;
+		this.dispatchEvent(new CustomEvent('rr-select', {
+			bubbles: true,
+			composed: true,
+		}));
```
Comment (tdjager) [inline src/components/lists-and-menus/menu/rr-menu.ts:56]: **IMPORTANT: Menu stays open after item selection**

`_handleClick` dispatches the `rr-select` event but never closes the menu. Standard menu UX expects the menu to dismiss after an item is selected. The Popover API's light dismiss only triggers on *outside* clicks, not on clicks within the popover.

**Fix:** After dispatching the event, close the parent menu:
```ts
this.closest('rr-menu')?.hidePopover();
```

File: src/components/control-groups/toolbar/rr-toolbar.styles.ts [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+		max-width: 100%;
+	}
+
+	:host([size="md"]) .toolbar__subtitle,
+	:host(:not([size])) .toolbar__subtitle {
+		font: var(--primitives-font-body-xs-regular-flat);
+	}
+
+	:host([size="sm"]) .toolbar__subtitle {
+		font-size: 12px;
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.styles.ts]: **IMPORTANT: Hardcoded pixel values violate design token rules**

Lines 253-255 use hardcoded `font-size: 12px; font-weight: 400; line-height: 1.125` instead of design tokens. The project rules explicitly state: "Only design tokens — never hardcode." The md variant on line 249 correctly uses `var(--primitives-font-body-xs-regular-flat)`.

Also note lines 230 and 235: `font-weight: 550` overrides the weight from the font shorthand token. If 550 is the intended weight, consider using the existing token `var(--primitives-font-weight-body-bold)` which is defined as 550 in settings.css.

**Fix:** Use an appropriate font token for the sm subtitle.

File: src/components/control-groups/toolbar/rr-toolbar.test.ts [hunk 7]
```diff
@@ -1,18 +1,88 @@
...
+		const label = el.shadowRoot?.querySelector('.toolbar__item-label');
+		expect(label).not.toBeNull();
+		const style = getComputedStyle(label!);
+		expect(style.display).toBe('none');
+	});
+	it('registers marker elements as custom elements', () => {
+		expect(customElements.get('rr-toolbar-item')).toBeDefined();
+		expect(customElements.get('rr-toolbar-title-group')).toBeDefined();
+	});
 });
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.test.ts:318]: **IMPORTANT: Toolbar tests lack coverage for core logic**

The toolbar is a 567-line component with complex overflow management, priority-based collapsing, resize/mutation observation, and menu synchronization. The test file has only 7 basic attribute/smoke tests. None of the following are covered:

- Overflow behavior (the core feature of this PR)
- Priority-based item collapsing order
- Menu creation and item synchronization
- Pinned overflow items
- Fluid item behavior
- ResizeObserver / MutationObserver integration

Per project rules: "Logic tests (verplicht bij complexe logica): test MutationObservers, slot management, attribuut propagatie, event handlers, state transitions."

**Fix:** Add tests for overflow mechanism, priority ordering, and menu sync at minimum.

File: src/components/lists-and-menus/menu/rr-menu.test.ts [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+	it('renders without error', async () => {
+		el = await fixture('<rr-menu-item></rr-menu-item>');
+		await waitForUpdate(el);
+		expect(el.shadowRoot).not.toBeNull();
+	});
+
+	it('reflects title attribute', async () => {
+		el = await fixture('<rr-menu-item text="Bewerk"></rr-menu-item>');
+		await waitForUpdate(el);
+		expect(el.getAttribute('title')).toBe('Bewerk');
```
Comment (tdjager) [inline src/components/lists-and-menus/menu/rr-menu.test.ts]: **IMPORTANT: Wrong attribute checked in assertion**

This test checks `el.getAttribute('title')` but the component property is `text` (with `reflect: true`), which reflects to the `text` attribute — not `title`. The HTML `title` attribute is the native tooltip attribute, which is unrelated.

**Fix:**
```ts
it('reflects text attribute', async () => {
    el = await fixture('<rr-menu-item text="Bewerk"></rr-menu-item>');
    await waitForUpdate(el);
    expect(el.getAttribute('text')).toBe('Bewerk');
});
```

File: src/assets/css/settings.css [hunk 9]
```diff
@@ -248,19 +248,19 @@
...
-	--primitives-font-size-200: 20px;
-	--primitives-font-size-300: 23px;
-	--primitives-font-size-400: 26px;
-	--primitives-font-size-500: 29px;
-	--primitives-font-size-600: 32px;
-	--primitives-font-size-700: 36px;
-	--primitives-font-size-800: 41px;
-	--primitives-font-size-900: 46px;
-	--primitives-font-size-1000: 52px;
+	--primitives-font-size-70: 0.79rem;    /* 12px */
```
Comment (tdjager) [inline src/assets/css/settings.css]: **IMPORTANT: Font-size rem values don't match their px comments**

These values appear to use a modular type scale rather than exact px-to-rem conversions:
- `0.79rem` = 12.64px (comment says 12px; exact would be `0.75rem`)
- `0.889rem` = 14.224px (comment says 14px; exact would be `0.875rem`)

The scale ratio (~1.125) is consistent, so this may be intentional. But if so, the `/* px */` comments are misleading and should be updated to reflect actual rendered sizes. This is also a breaking change from the previous px-based values.

**Fix:** Either use exact rem conversions (`0.75rem`, `0.875rem`, etc.) or update the comments to show actual px values.

File: src/components/control-groups/toolbar/rr-toolbar.ts [hunk 10]
```diff
@@ -1,118 +1,566 @@
...
+		) {
+			this._syncMenuItems();
+			this._syncMenuAnchor();
+			if (!this._hasMeasured) {
+				this.updateComplete.then(() => this._updateAreaVars());
+			}
+		}
+	}
+
+	private _createMenu(): void {
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.ts:188]: **IMPORTANT: Add guard against double menu creation**

`_createMenu()` appends an `rr-menu` to `document.body`. While `disconnectedCallback` removes it, if `connectedCallback` fires twice without a disconnect in between (possible in some framework edge cases), a second menu element will leak.

**Fix:** Add a guard at the top:
```ts
private _createMenu(): void {
    if (this._menu) return;
    // ...
}
```

File: src/components/control-groups/toolbar/rr-toolbar.ts [hunk 11]
```diff
@@ -1,118 +1,566 @@
...
+
+	private _syncMenuItems(): void {
+		if (!this._menu) return;
+		this._menu.innerHTML = '';
+
+		const endItems = this._endChildren.filter((c): c is Extract<ToolbarChild, { type: 'item' }> => c.type === 'item');
+		const startItems = this._startChildren.filter((c): c is Extract<ToolbarChild, { type: 'item' }> => c.type === 'item');
+		const centerItems = this._centerChildren.filter((c): c is Extract<ToolbarChild, { type: 'item' }> => c.type === 'item');
+
+		const prioritized = [
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.ts]: **IMPORTANT: Duplicated priority sorting logic**

This sorting logic (lines 173-184) is nearly identical to `_getPrioritizedItems()` (lines 235-245), with one subtle difference: this version adds `.reverse()` at line 184. This DRY violation means changes to sorting behavior must be made in two places.

**Fix:** Refactor to reuse `_getPrioritizedItems()`:
```ts
const prioritized = [...this._getPrioritizedItems()].reverse();
```

File: src/components/lists-and-menus/menu/rr-menu.ts (hunk 1 above)
Comment (bartvandebiezen) [reply src/components/lists-and-menus/menu/rr-menu.ts]: renamed menu bar item

File: src/components/control-groups/toolbar/rr-toolbar.ts [hunk 12]
```diff
@@ -1,118 +1,561 @@
...
+	private _buildChildrenForArea(areaTag: string): ToolbarChild[] {
+		const areaEl = this.querySelector(areaTag);
+		if (!areaEl) return [];
+
+		return Array.from(areaEl.children).map(el => {
+			const tag = el.tagName.toLowerCase();
+
+			if (tag === 'rr-toolbar-title-group') {
+				const id = this._getId(el);
+				if (el.parentElement !== this) this.appendChild(el);
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.ts:469]: **Review finding:** `_buildChildrenForArea` moves children from area containers to the toolbar root via `this.appendChild(el)`. This destructively alters consumer markup — area elements end up empty after initialization. Additionally, because children are moved out of the area subtree, the MutationObserver on area elements (set up with `attributes: true, subtree: true`) won't fire for attribute changes on those moved children. Dynamically changing `label`, `priority`, `min-width`, or `width` on toolbar items after initial render will be silently ignored.

File: src/components/navigation/menu-bar/rr-menu-bar.styles.ts [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+
+	.menu-bar-item {
+		appearance: none;
+		border: none;
+		margin: 0;
+		background: none;
+		text-decoration: none;
+		display: flex;
+		position: relative;
+		height: 44px;
```
Comment (tdjager) [inline src/components/navigation/menu-bar/rr-menu-bar.styles.ts:29]: **Review finding:** Hardcoded values violate the project convention of using design tokens. This `height: 44px` should be `var(--semantics-controls-m-min-size)`. Also: `min-width: 200px` (line 234) and `z-index: 10000` (line 238) in the overflow dropdown are hardcoded magic numbers.

File: src/components/navigation/menu-bar/rr-menu-bar.ts [hunk 14]
```diff
@@ -1,572 +1,417 @@
...
+
+			for (let i = overflowStartIndex; i < items.length; i++) {
+				items[i].style.display = 'none';
+				items[i].setAttribute('data-overflow', 'true');
+
+				const dropdownItem = document.createElement('button');
+				dropdownItem.className = 'overflow-item';
+				dropdownItem.setAttribute('role', 'menuitem');
+				dropdownItem.setAttribute('tabindex', '-1');
+				dropdownItem.textContent = items[i].textContent;
```
Comment (tdjager) [inline src/components/navigation/menu-bar/rr-menu-bar.ts:281]: **Review finding:** When menu-bar items overflow into the dropdown, only `textContent` is copied to plain `<button>` elements. This loses any rich content (icons, badges) and accessibility attributes from the original `rr-menu-bar-item`. Consider cloning or re-rendering items to preserve their full structure.

File: src/components/navigation/menu-bar/rr-menu-bar.template.ts [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+export function template(this: RRMenuBar) {
+	return html`
+		<div
+			class="container"
+			part="container"
+		>
+			<slot name="title"></slot>
+			<nav
```
Comment (tdjager) [inline src/components/navigation/menu-bar/rr-menu-bar.template.ts:65]: **Review finding:** The `<nav>` element has `role="none"` which strips its navigation semantics. There is no `role="menubar"` on the container and no `role="menuitem"` on the items. The keyboard navigation implements menubar patterns (ArrowLeft/Right) but without the corresponding ARIA roles, screen readers cannot identify this as a navigation menu bar. WCAG 2.1 AA concern.

File: src/components/navigation/menu-bar/rr-menu-bar.ts [hunk 16]
```diff
@@ -1,572 +1,417 @@
...
+
+	@property({ type: String })
+	href = '';
+
+	@property({ type: Boolean, reflect: true })
+	disabled = false;
+
+	override connectedCallback(): void {
+		super.connectedCallback();
+		this.setAttribute('role', 'none');
```
Comment (tdjager) [inline src/components/navigation/menu-bar/rr-menu-bar.ts:22]: **Review finding:** `RRMenuBarItem` sets `role="none"` on the host and does not use `delegatesFocus`. When keyboard navigation calls `focus()` on this element (line 184), focus won't reach the inner `<button>` or `<a>` in shadow DOM. Fix: add `static shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true }` to `RRMenuBarItem`, or change the keyboard handler to focus the inner element directly.

File: src/components/control-groups/toolbar/rr-toolbar.ts [hunk 17]
```diff
@@ -1,118 +1,561 @@
...
+		if (!this._menu) return;
+		this._menu.innerHTML = '';
+
+		const prioritized = [...this._getPrioritizedItems()].reverse();
+
+		prioritized.forEach(child => {
+			if (!this._overflowIds.has(child.id)) return;
+			if (child.overflowItems.length === 0) return;
+			child.overflowItems.forEach(el => {
+				const clone = el.cloneNode(true) as Element;
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.ts:224]: **Review finding:** `cloneNode(true)` does not copy event listeners added via `addEventListener`. Consumer-attached click handlers or other listeners on `rr-menu-item` elements will be silently lost in the overflow menu clone. Consider documenting this limitation, or using slot projection / moving original elements instead of cloning.

File: src/components/control-groups/toolbar/rr-toolbar.template.ts [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+				${hasCenterChildren ? html`
+					${leftSpacerZero ? nothing : html`<div class="toolbar__left-spacer"></div>`}
+					${renderChildren(centerChildren, allChildren, overflowIds)}
+					${rightSpacerZero ? nothing : html`<div class="toolbar__right-spacer"></div>`}
+				` : isSoloFluidItem ? nothing : html`
+					<div class="toolbar__flexible-spacer"></div>
+				`}
+				${renderChildren(endChildren, allChildren, overflowIds)}
+			</div>
+			<div class="toolbar__overflow-button ${hasOverflow ? '' : 'is-hidden'}">
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.template.ts:116]: **Review finding:** The overflow button has no explicit `@click` handler — menu opening relies on `rr-menu`'s internal document-level click detection. This is fragile implicit coupling. Also missing `aria-controls` attribute pointing to the menu's ID (WCAG 4.1.2). Consider adding an explicit click handler and `aria-controls`.

File: src/components/navigation/menu-bar/rr-menu-bar.template.ts [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+		<path d="M11.9995 15.1715L3.4 6.59998L2 7.99998L11.9995 18L22 7.99998L20.6 6.59998L11.9995 15.1715Z"/>
+	</svg>
+`;
+
+export function menuBarItemTemplate(this: RRMenuBarItem) {
+	const safeHref = this._sanitizeUrl(this.href);
+	const isLink = Boolean(safeHref);
+	const tabindex = this.disabled ? '-1' : '0';
+
+	if (isLink) {
```
Comment (tdjager) [inline src/components/navigation/menu-bar/rr-menu-bar.template.ts:21]: **Review finding:** When `rr-menu-bar-item` has an `href` and is disabled, `aria-disabled` does NOT prevent navigation — the browser will still follow the link on click. The `_handleClick` handler calls `preventDefault()` on the host element, but for shadow DOM `<a>` elements the default action may fire before the event reaches the host. Keyboard activation (Enter) also bypasses `pointer-events: none`. Fix: remove the `href` attribute when disabled, or attach the click handler directly to the `<a>` element via `@click`.

**Review by tdjager — COMMENTED** [review]
Review - 8 findings inline

**Review by tdjager — CHANGES_REQUESTED** [review]

File: src/components/navigation/menu-bar/rr-menu-bar.styles.ts (hunk 13 above)
Comment (bartvandebiezen) [reply src/components/navigation/menu-bar/rr-menu-bar.styles.ts:29]: This was already the case before the pull request. Will update it when I refactor the menu-bar.

File: src/components/navigation/menu-bar/rr-menu-bar.template.ts (hunk 15 above)
Comment (bartvandebiezen) [reply src/components/navigation/menu-bar/rr-menu-bar.template.ts:65]: This was already the case before the pull request. Will update it when I refactor the menu-bar.

File: src/components/navigation/menu-bar/rr-menu-bar.template.ts (hunk 19 above)
Comment (bartvandebiezen) [reply src/components/navigation/menu-bar/rr-menu-bar.template.ts:21]: This was already the case before the pull request. Will update it when I refactor the menu-bar.

File: src/components/navigation/menu-bar/rr-menu-bar.ts (hunk 14 above)
Comment (bartvandebiezen) [reply src/components/navigation/menu-bar/rr-menu-bar.ts:281]: This was already the case before the pull request. Will update it when I refactor the menu-bar.

File: src/components/navigation/menu-bar/rr-menu-bar.ts (hunk 16 above)
Comment (bartvandebiezen) [reply src/components/navigation/menu-bar/rr-menu-bar.ts:22]: This was already the case before the pull request. Will update it when I refactor the menu-bar.

File: src/components/control-groups/toolbar/rr-toolbar.ts [hunk 20]
```diff
@@ -1,118 +1,590 @@
...
+	}
+
+	private _handleOverflowButtonClick(): void {
+		if (!this._menu) return;
+		if (this._menuOpen) {
+			(this._menu as unknown as { hidePopover: () => void }).hidePopover();
+		} else {
+			(this._menu as unknown as { showPopover: () => void }).showPopover();
+		}
+	}
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.ts:186]: **Double-toggle race condition**: When clicking the overflow button, this handler calls `showPopover()`, which synchronously fires a `toggle` event setting `_isOpen = true` on the menu. Then the menu's `_handleDocumentClick` (registered on `document`) also fires for the same click — it sees the anchor in the event path, checks `_isOpen` (now `true`), and immediately calls `hidePopover()`. The menu flashes open then closes.

Fix: either call `event.stopPropagation()` here to prevent the click from reaching the document listener, or have `_handleDocumentClick` in `rr-menu` skip acting when `anchorElement` is set (since the caller controls toggling).

File: src/components/control-groups/toolbar/rr-toolbar.stories.js [hunk 21]
```diff
@@ -1,60 +1,806 @@
...
+		>
+			<rr-toolbar-start-area>
+				<rr-toolbar-item label="Vet">
+					<rr-icon-button>
+						<rr-icon name="bold"></rr-icon>
+						Vet
+					</rr-icon-button>
+					<rr-menu-item
+						text="Vet"
+						selectable
```
Comment (tdjager) [inline src/components/control-groups/toolbar/rr-toolbar.stories.js]: `selectable` is not a defined attribute on `rr-menu-item`. The component has `type` (`checkbox`/`radio`) and `selected` (boolean). This attribute is silently ignored, so stories demonstrate non-functional behavior.

Fix: replace `selectable` with `type="checkbox"` (used in 27 places across this file).

**Review by tdjager — CHANGES_REQUESTED** [review]
Two issues that need fixing before merge.

**Review by tdjager — APPROVED** [review]
Approved, still a few things to note:  

1. No toolbar keyboard navigation — role="toolbar" without ArrowLeft/ArrowRight handling (WCAG a11y gap)
  2. Hardcoded pixel values in rr-menu-bar.styles.ts — 44px, 16px, 200px, z-index: 10000 should use tokens
  3. Missing menu tests — Escape key, divider visibility, popover toggle flow untested
  4. Empty PR description — breaking changes (rr-menu-item API, toolbar area wrappers, import paths) undocumented
